types/bitcoinjs-lib/index.d.ts

```diff
@@ -1,114 +1,234 @@
+
+    static fromBase58(string: string, networks?: any): HDNode;
+
+    static fromSeedBuffer(seed: Buffer, network: Network, ...args: any[]): HDNode;
+
+    static fromSeedHex(hex: string, network: Network): HDNode;
+
 }
+
 export class Transaction {
+    version: number;
+    locktime: number;
+    ins: Array<Input>;
+    outs: Array<Output>;
     constructor();
-    addInput(hash: any, index: any, sequence: any, scriptSig: any, ...args: any[]): any;
+
+    addInput(hash: Buffer, index: number, sequence: any, scriptSig: Buffer, ...args: any[]): any;
+
     addOutput(scriptPubKey: any, value: any, ...args: any[]): any;
-    byteLength(): number;
+
+    byteLength(): any;
+
     clone(): Transaction;
-    getHash(): any;
+
+    getHash(): Buffer;
+
     getId(): any;
-    hashForSignature(inIndex: any, prevOutScript: any, hashType: any, ...args: any[]): any;
+
+    hasWitnesses(): any;
+
+    hashForSignature(inindex: number, prevOutScript: any, hashType: any, ...args: any[]): any;
+
+    hashForWitnessV0(inindex: number, prevOutScript: any, value: any, hashType: any, ...args: any[]): any;
+
     isCoinbase(): boolean;
-    setInputScript(index: any, scriptSig: any, ...args: any[]): void;
-    toBuffer(buffer?: any, initialOffset?: any): Buffer;
-    toHex(): string;
+
+    setInputScript(index: number, scriptSig: Buffer, ...args: any[]): void;
+
+    setWitness(index: number, witness: any, ...args: any[]): void;
+
+    toBuffer(buffer: Buffer, initialOffset: any): Buffer;
+
+    toHex(): any;
+
+    static ADVANCED_TRANSACTION_FLAG: number;
+
+    static ADVANCED_TRANSACTION_MARKER: number;
+
     static DEFAULT_SEQUENCE: number;
+
     static SIGHASH_ALL: number;
+
     static SIGHASH_ANYONECANPAY: number;
+
     static SIGHASH_NONE: number;
+
     static SIGHASH_SINGLE: number;
-    static fromBuffer(buffer: any, __noStrict: any): Transaction;
-    static fromHex(hex: any): Transaction;
-    static isCoinbaseHash(buffer: any): Transaction;
+
+    static fromBuffer(buffer: Buffer, __noStrict: any): Transaction;
+
+    static fromHex(hex: string): Transaction;
+
+    static isCoinbaseHash(buffer: Buffer): any;
+
 }
+
 export class TransactionBuilder {
-    constructor(network?: Network);
-    addInput(txHash: any, vout: any, sequence?: any, prevOutScript?: any): any;
-    addOutput(scriptPubKey: any, value: any): any;
+    constructor(network: Network, maximumFeeRate?: any);
+
+  addInput(txhash: Buffer|string|Transaction, vout: number, sequence?: number, prevOutScript?: Buffer): void;
+
+  addOutput(scriptPubKey: Buffer|string, value: any): any;
+
     build(): Transaction;
+
     buildIncomplete(): Transaction;
-    setLockTime(locktime: number): any;
+
+    setLockTime(locktime: number): void;
+
     setVersion(version: number): void;
-    sign(index: any, keyPair: any, redeemScript?: any, hashType?: any): any;
+
+    sign(vin: number, keyPair: ECPair, redeemScript?: Buffer, hashType?: number, witnessValue?: any, witnessScript?: any): void;
```

Comment:
Parameter `witnessValue` should have type `number`.
Parameter `witnessScript` should have type `Buffer`.